build/source/engine/computFlux.f90

```diff
@@ -0,0 +1,851 @@
+USE multiconst,only:&
+                    gravity,      & ! acceleration of gravity              (m s-2)
+                    Tfreeze,      & ! temperature at freezing              (K)
+                    LH_fus,       & ! latent heat of fusion                (J kg-1)
+                    LH_vap,       & ! latent heat of vaporization          (J kg-1)
+                    LH_sub,       & ! latent heat of sublimation           (J kg-1)
+                    Cp_air,       & ! specific heat of air                 (J kg-1 K-1)
+                    iden_air,     & ! intrinsic density of air             (kg m-3)
+                    iden_ice,     & ! intrinsic density of ice             (kg m-3)
+                    iden_water      ! intrinsic density of liquid water    (kg m-3)
+
+! provide access to the derived types to define the data structures
+USE data_types,only:&
+                    var_i,        & ! data vector (i4b)
+                    var_d,        & ! data vector (dp)
+                    var_ilength,  & ! data vector with variable length dimension (i4b)
+                    var_dlength,  & ! data vector with variable length dimension (dp)
+                    model_options   ! defines the model decisions
+
+! look-up values for the choice of groundwater representation (local-column, or single-basin)
+USE mDecisions_module,only:       &
+ localColumn,                     & ! separate groundwater representation in each local soil column
+ singleBasin                        ! single groundwater store over the entire basin
+
+! look-up values for the choice of groundwater parameterization
+USE mDecisions_module,only:       &
+ qbaseTopmodel,                   & ! TOPMODEL-ish baseflow parameterization
+ bigBucket,                       & ! a big bucket (lumped aquifer model)
+ noExplicit                         ! no explicit groundwater parameterization
+
+! look-up values for the form of Richards' equation
+USE mDecisions_module,only:       &
+ moisture,                        & ! moisture-based form of Richards' equation
+ mixdform                           ! mixed form of Richards' equation
+
+! look-up values for the choice of boundary conditions for hydrology
+USE mDecisions_module,only:       &
+ prescribedHead,                  & ! prescribed head (volumetric liquid water content for mixed form of Richards' eqn)
+ funcBottomHead,                  & ! function of matric head in the lower-most layer
+ freeDrainage,                    & ! free drainage
+ liquidFlux,                      & ! liquid water flux
+ zeroFlux                           ! zero flux
+
+implicit none
+private
+public::computFlux
+public::soilCmpres
+! control parameters
+real(dp),parameter  :: valueMissing=-9999._dp     ! missing value
+real(dp),parameter  :: verySmall=tiny(1.0_dp)     ! a very small number
+real(dp),parameter  :: veryBig=1.e+20_dp          ! a very big number
+real(dp),parameter  :: dx = 1.e-8_dp              ! finite difference increment
```

Comment:
The user does have control over most algorithmic control parameters. There are some -- like the finite difference increment -- where we may not want to give the user control as this can cause problems if set incorrectly.